src/review-form.ts

```diff
@@ -37,41 +41,49 @@ export class ReviewForm extends LitElement {
   @property({ type: Object }) oldReview?: Review;
 
   /* Errors to add to the form, if any */
-  @property({ type: Array }) errors?: string[];
+  @property({ type: Array }) errors: string[] = [];
+
+  /* Service for activating the recaptcha challenge */
+  @property({ type: Object }) recaptchaManager?: RecaptchaManagerInterface;
+
+  /* Recaptcha widget for the form */
+  @property({ type: Object }) recaptchaWidget?: RecaptchaWidgetInterface;
+
+  /* Token generated by recaptcha process */
+  @state()
+  private recaptchaToken: string = '';
 
   /* Number of stars currently selected */
   @state()
   private currentStars: number = 0;
 
+  /* The form to be submitted */
+  @query('#review-form')
+  private reviewForm!: HTMLFormElement;
+
   render() {
-    return html`<form action="${this.baseHost}${this.endpointPath}">
-      ${this.errors
+    return html`<form
+      id="review-form"
+      action="${this.baseHost}${this.endpointPath}"
+      method="post"
+    >
+      ${this.errors.length
         ? html`<div class="errors">${this.errors.join(' ')}</div>`
         : nothing}
       ${this.starsInputTemplate} ${this.subjectInputTemplate}
-      ${this.bodyInputTemplate}
-      ${this.identifier
-        ? html`<input
-            type="hidden"
-            name="identifier"
-            .value=${this.identifier}
-          />`
-        : nothing}
-      ${this.token
-        ? html`<input
-            type="hidden"
-            name="field_reviewtoken"
-            .value=${this.token}
-          />`
-        : nothing}
+      ${this.bodyInputTemplate} ${this.hiddenInputsTemplate}
       ${this.actionButtonsTemplate}
     </form>`;
   }
 
-  protected updated(changed: PropertyValues): void {
+  protected async updated(changed: PropertyValues): Promise<void> {
```

Comment:
`updated` gets called by the Lit framework and it doesn't treat it as asynchronous so even though you're decorating this with `async` and `Promise`, the caller is never actually waiting for it to complete.

I would probably leave the signature to `protected updated() ... : void` for consistency. Then maybe do something like 
```
if (changed.has('recaptchaManager') && this.recaptchaManager) {
  this.setupRecaptcha()
}

...

private async setupRecaptcha(): Promise<void> {
   ...
}
```

This at least lets us know within our code that we're doing asynchronous stuff (even though you're not waiting for it in `updated`. It's a bit of a weird flow, but we can't really change how `Lit` works for these methods.